### ISSUE TITLE
X-<Header> converts to lowercase

### ISSUE THREAD

**Issue by crusty-dave** [actix/actix-web#396]
With the following code:

`            Ok(HttpResponse::Ok()
                .header("X-Subject-Token", token.token)
                .body(""))
`

It transmits "X-Subject-Token" as "x-subject-token", this would seem to be a bug, it shouldn't be changing any header tags like this AFAIK?

**Comment by crusty-dave** [comment]
Per RFC 7231, the headers should be case-insensitive, but I would still expect actix-web to honor the case that I have provided.

https://tools.ietf.org/html/rfc7230#page-19

> 3.2.  Header Fields
> 
>    Each header field consists of a case-insensitive field name followed
>    by a colon (":"), optional leading whitespace, the field value, and
>    optional trailing whitespace.
> 

Note that the HTTP RFCs use camel-case with dash (-) separators when they describe the fields.

This is the first time that I have witnessed something like "content-type" instead of "Content-Type".
